docs/amdgpu_kernel_optimization_guide.md

```diff
@@ -201,17 +208,29 @@ completes.
 #### Avoiding LDS Bank Conflicts
 
 With the number of LDS banks (32) not matching the subgroup size (64) nor the
-SIMD size (16), it's not immediately obvious when bank conflicts arise.
-
-On the hardware level, LDS services request from either pair of neighboring
-SIMDs per cycle. However, because it's very hard to account for what code may
-be executing on the neighboring SIMD, we developed a mental model that analyzes
-execution of a single subgroup only. We determine the possible rate of bank
-conflicts by looking at banks accessed by each group of 16 adjacent threads
-executing on the same SIMD.
+SIMD size (16), it is not immediately obvious when bank conflicts arise.
+
+LDS is able to access 32 banks per cycle. Depending on the exact LDS instruction
+used (read/write of `b32` vs. `b64` vs. `b128`), different number of threads
+within the same subgroup access LDS banks. The size of this group of threads is
+determined by the number of dwords accessed by each thread. Assuming no bank
+conflicts, a `b32` access ( single dword) handles 32 adjacent threads (amounting
+to 32 VGPRs), while for `b128` *all* VGPRs of 8 adjacent threads are handled
```

Comment:
For `b128`, are we handling the first 8 threads in one-go first, and then the next 8 threads, etc? Would be nice to clear on this.